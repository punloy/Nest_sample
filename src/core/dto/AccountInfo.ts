export class AccountInfo {
    id: number;
    name: string
}
