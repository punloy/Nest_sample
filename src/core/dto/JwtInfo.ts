export class JwtInfo {
    id: number;
    data: {
        name: string,
    };
    jti?: string;
}
