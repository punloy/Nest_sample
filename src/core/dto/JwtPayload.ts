export class JwtPayload {
    id: number;
    data: {
        name: string,
    };
    jti?: string;
}
