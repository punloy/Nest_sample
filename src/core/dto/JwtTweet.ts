export class JwtTweet {
    id: number;
    data: {
        name: string,
        tittle: string,
        content: string
    };
    jti?: string;
}
