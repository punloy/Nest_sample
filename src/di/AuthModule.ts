import { Module } from '@nestjs/common';
import { PassportModule } from '@nestjs/passport';
import { TypeOrmModule } from '@nestjs/typeorm';
import { JwtStrategy } from '../core/jwt/JwtStrategy';
import { LocalUsertrategy } from '../core/jwt/LocalUserStrategy';
import { tweet } from '../core/index';
import { AuthService } from '../service/AuthService';

@Module({
    imports: [
        PassportModule,
        TypeOrmModule.forFeature(Object.values(tweet))
    ],
    providers: [
        AuthService,
        LocalUsertrategy,
        JwtStrategy,
    ]
})
export class AuthModule { }