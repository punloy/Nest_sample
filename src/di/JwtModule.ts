import { Module } from "@nestjs/common";
import { JwtService } from "../service/JwtService";

@Module({
    providers: [JwtService],
    exports: [JwtService]
})
export class JwtModule { }
