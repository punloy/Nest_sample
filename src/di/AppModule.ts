import { Module, ValidationPipe } from '@nestjs/common';
import { HealthModule } from './HealthModule';
import { APP_PIPE, APP_FILTER } from '@nestjs/core';
import config from "config";
import { TypeOrmModuleOptions } from "@nestjs/typeorm";
import { TweetDatabaseModule } from './TweetDatabaseModule';
import { HttpExceptionFilter } from '../core/filter/HttpExceptionFilter';
import { UserModule } from './UserModule';
import { AuthModule } from './AuthModule';
import { TweetModule } from './TweetModule';

const tweetOption: Partial<TypeOrmModuleOptions> = config.get("tweetDB");

@Module({
  imports: [
    TweetDatabaseModule.register(tweetOption),
    UserModule,
    TweetModule,
    AuthModule,
    HealthModule],
  providers: [
    {
      provide: APP_PIPE,
      useClass: ValidationPipe
    }, {
      provide: APP_FILTER,
      useClass: HttpExceptionFilter
    }
  ],
})
export class AppModule { }
