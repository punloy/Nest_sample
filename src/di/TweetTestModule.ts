import { DynamicModule, Module } from "@nestjs/common";
import { TypeOrmModuleOptions } from "@nestjs/typeorm";
import config from "config";
import { AppModule } from "./AppModule";
import { TweetDatabaseModule } from "./TweetDatabaseModule";

const tweetOptions: Partial<TypeOrmModuleOptions> = config.get("tweetDB");

@Module({})
export class TweetTestModule {
    public static register(): DynamicModule {
        return {
            module: TweetTestModule,
            imports: [
                TweetDatabaseModule.register(tweetOptions),
                AppModule
            ]
        };
    }
}
