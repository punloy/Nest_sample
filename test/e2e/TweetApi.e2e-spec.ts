import { JwtPayload } from "../../src/core/dto/JwtPayload";
import { TweetTestModule } from "../../src/di/TweetTestModule";
import { TestTweetData } from "../helper";
import { withJwt, withTestApp } from "../helper/TestApp";

const imports = [TweetTestModule.register()];

describe("TweetApi (e2e)", () => {
    let jwtpayload1, jwtpayload2: JwtPayload;

    beforeAll(withTestApp(imports, async () => {
        jwtpayload1 = await TestTweetData.createTestUser({ userId: 1 });
        jwtpayload2 = await TestTweetData.createTestUser({ userId: 2 });
    }));

    it("/tweet/add (POST) should 200", withTestApp(imports, async (app, request) => {
        withJwt(request, { id: jwtpayload1.id, data: jwtpayload1.data, jti: jwtpayload1.jti })
        const result = await request
            .post("/api/v1/tweet/add")
            .send({
                tittle: "tweet tittle",
                content: "content"
            })
            .expect(200);
        expect(result.body).toMatchSnapshot();
    }))

    it("/tweet/add (POST) should 403 1", withTestApp(imports, async (app, request) => {
        withJwt(request, { id: jwtpayload2.id, data: jwtpayload2.data, jti: jwtpayload2.jti })
        const result = await request
            .post("/api/v1/tweet/add")
            .send({
                tittle: "",
                content: "content"
            })
            .expect(403);
        expect(result.body).toMatchSnapshot();
    }))

    it("/tweet/add (POST) should 403 3", withTestApp(imports, async (app, request) => {
        withJwt(request, { id: jwtpayload2.id, data: jwtpayload2.data, jti: jwtpayload2.jti })
        const result = await request
            .post("/api/v1/tweet/add")
            .send({
                tittle: "test tittle",
                content: ""
            })
            .expect(403);
        expect(result.body).toMatchSnapshot();
    }))
})