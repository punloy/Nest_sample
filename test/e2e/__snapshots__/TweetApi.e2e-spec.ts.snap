// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TweetApi (e2e) /tweet/add (POST) should 200 1`] = `
Object {
  "data": Object {
    "content": "content",
    "name": "test user 1",
    "tittle": "tweet tittle",
  },
  "message": "success",
}
`;

exports[`TweetApi (e2e) /tweet/add (POST) should 403 1 1`] = `
Object {
  "data": Object {},
  "message": "tittle should not empty",
}
`;

exports[`TweetApi (e2e) /tweet/add (POST) should 403 3 1`] = `
Object {
  "data": Object {},
  "message": "content should not empty",
}
`;

exports[`TweetApi (e2e) /tweet/list (GET) should 200 1`] = `
Object {
  "data": Object {
    "tweetList": Array [
      Object {
        "content": "content 1",
        "tittle": "tittle 1",
      },
      Object {
        "content": "content",
        "tittle": "tweet tittle",
      },
    ],
  },
  "message": "success",
}
`;
